larq/callbacks.py

```diff
@@ -4,21 +4,38 @@
 class HyperparameterScheduler(tf.keras.callbacks.Callback):
     """Generic hyperparameter scheduler.
 
+    !!! example
+        ```python
+        optimizer = lq.optimizers.CaseOptimizer(
+            (
+                lq.optimizers.Bop.is_binary_variable,
+                lq.optimizers.Bop(threshold=1e-6, gamma=1e-3),
+            ),
+            default_optimizer=tf.keras.optimizers.Adam(0.01),
+        )
+        callbacks = [
+            HyperparameterScheduler(
+                lambda x:0.001*(0.1**(x//30)), 'gamma', optimizer.optimizers[0]
+            )
+        ]
+        ```
     # Arguments
+    optimizer: the optimizer that contains the hyperparameter that will be scheduled.
```

Comment:
I think this should be optional (if not a `CaseOptimizer`), since most of the time you'd like to schedule parameters from `model.optimizer`. And this can lead to tricky bugs where you schedule an hyperparameter from an optimizer that is not actually used in your model.

Reply:
Alright, I agree that this can lead to very hard to debug bugs (I had one of those myself) so I guess making this optional will both give an example how to use it properly and avoid bugs in the already existing code. I will make it optional.

Reply:
I made it optional but did not yet add a test for the case that no optimizer argument is given. Do you think it would be good to have one?